src/support/bits.cpp

```diff
@@ -163,6 +163,9 @@ bool isPowerOf2Float(float v) {
   // unmasked input value. If they are equal, our value is a power of
   // two. Also, we reject all values which are less than the minimal possible
   // power of two or greater than the maximum possible power of two.
+  // We check values only with exponent in more limited ranges
+  // [-126..+126] for floats and [-1022..+1022] for doubles for avoiding
+  // overflows and reject NaNs, infinity and denormals.
```

Comment:
We talked with @kripken about this [changes here](https://github.com/WebAssembly/binaryen/pull/3018#discussion_r505122515). Is this enough for extra description?

Reply:
Looks good to me, but I want @kripken to sign off on it too, since that previous discussion was with him.

Reply:
IIUC, this function rejects an exponent of 1023 even though it is not a NaN, infinity, or a denormal? It rejects it only because it is not possible to do `1/x` and get a valid double, since the valid range is not symmetric, `−1022 .. +1023`.

That's worth a comment as it is an odd property of the function, IIUC. Maybe something like:
```
We also reject "asymmetric exponents", like +1023, because the range of (non-NaN, non-infinity) values is -1022..+1023, and it is convenient in optimizations to depend on being able to invert a power of two without losing precision.
```

Also it is depended on in OptimizeInstructions. Please also comment in that pass in the place where it depends on this.

Or, if the name were `isPowerOfTwoInvertibleFloat` that might be clearer?

Reply:
> We also reject "asymmetric exponents", like +1023, because the range of (non-NaN, non-infinity) values is -1022..+1023, and it is convenient in optimizations to depend on being able to invert a power of two without losing precision

Sonds good. I'll add this

> Also it is depended on in OptimizeInstructions. Please also comment in that pass in the place where it depends on this.

Sure!

> Or, if the name were isPowerOfTwoInvertibleFloat that might be clearer?

Not sure about this.I guess `1023` isn't useful in any case not only for invertible.

Reply:
Why isn't 1023 useful? It's a valid power of 2, `​2^1023`. It's not a NaN or Infinity or anything special,

https://float.exposed/0x7fe0000000000000

Reply:
I agree, it's valid power of two value. So you suggest rename this function as `isPowerOfTwoInvertibleFloat`?

Reply:
Yes, the more I think about it, I think that's the right solution. Let's rename it to `isPowerOfTwoInvertibleFloat` and then there is less surprising stuff, and less to explain in comments.

Reply:
done!)